waspc/data/Generator/templates/react-app/src/queries/index.js

```diff
@@ -0,0 +1,18 @@
+import { useQuery as rqUseQuery } from 'react-query'
+
+export const useQuery = (queryFn, queryFnArgs, config) => {
+  if (queryFn === 'undefined') {
+    throw new Error('useQuery requires queryFn to be specified.')
+  }
+  if (!queryFn.useQueryKey) {
+    throw new Error('queryFn needs to have useQueryKey property defined.')
+  }
+
+  const rqResult = rqUseQuery({
+    queryKey: [queryFn.useQueryKey, queryFnArgs],
+    queryFn: (_key, args) => queryFn(args),
```

Comment:
`key` is not used, that is why it is underscored?

Reply:
Yes! I was inspired by Haskell here.